Move UTxO to Core

TimSheard opened the pull request:
Split the UTxO file in Shelley into two parts.
One that deals just with UTxO  Cardano.Ledger.UTxO and the othet which deals with what happens in the 
Shelley era, Cardano.Ledger.Shelley.UTxO

lehins commented on eras/shelley/test-suite/src/Test/Cardano/Ledger/Shelley/Generator/Core.hs:
```diff
@@ -109,17 +109,17 @@ import qualified Cardano.Ledger.Shelley.Tx as Ledger
+import Cardano.Ledger.UTxO
+  ( UTxO,
+    pattern UTxO,
+  )
```
Let's fix this while we at it: 
```suggestion
import Cardano.Ledger.UTxO (UTxO(UTxO))
```

TimSheard replied:
done

lehins reviewed (approved):
This is great. Just one minor recommendation

TimSheard committed:
copied, moved then added all UTxO.hs UTxO1.hs UTxO2.hs

TimSheard committed:
moved UTxO1.hs into core

TimSheard committed:
movedUTxO2.hs to Shelley UTxO.hs

TimSheard committed:
Split Cardano.Ledger.Shelley.UTxO into two different modules
Cardano.Ledger.UTxO that deals with just the UTxO was moved to core
Cardano.Ledger.Shelley.UTxO that deals with Shelley stuff
Both files maintain their git history from the original Cardano.Ledger.Shelley.UTxO
For backwards compatibility The Shelley version exports the core version. We can
    change this if we need to.

TimSheard commented:
After some magic, managed to restore the git history to both UTxO filles.

lehins reviewed (approved):
Very nice!!!

JaredCorduan commented on libs/cardano-ledger-core/src/Cardano/Ledger/UTxO.hs:
```diff
@@ -0,0 +1,251 @@
+  WitVKey (asWitness $ vKey keys) (coerce $ signedDSIGN @c (sKey keys) (extractHash safe))
+
+-- | Create witnesses for transaction
+makeWitnessesVKey ::
```
this is just a side comment while I'm thinking about it, definitely ignore for now in this PR. but I think lots of the functions in this module are not really that related to the `UTxO` and should be in their own module.

TimSheard replied:
For the next cleanup PR

JaredCorduan reviewed (approved):
fantastic, thank you @TimSheard !

lehins reviewed (changes requested):
@TimSheard Squashing all of the commits together erased all of the work you did in the attempt to preserve git history. Are you sure that is what you wanted?

lehins reviewed (approved):
:+1:
